src/command_modules/azure-cli-acr/azure/cli/command_modules/acr/import.py

```diff
@@ -35,40 +33,36 @@ def acr_import(cmd,
     _, resource_group_name = validate_managed_registry(
         cmd, registry_name, resource_group_name, IMPORT_NOT_SUPPORTED)
 
+    ImportImageParameters, ImportSource, ImportMode = cmd.get_models(
+        'ImportImageParameters', 'ImportSource', 'ImportMode')
+
     if not source:
         raise CLIError(INVALID_SOURCE_IMAGE)
-    source_image = source
 
-    slash = source.find('/')
+    import re
+    ACR = re.match(r'[0-9a-zA-Z/]+(\.azurecr\.)(io|cn|us|de)', source)
 
-    ImportImageParameters, ImportSource, ImportMode = cmd.get_models(
-        'ImportImageParameters', 'ImportSource', 'ImportMode')
-    if slash < 0:
-        if not source_registry:
-            from knack.prompting import prompt, NoTTYException
-            try:
-                source_registry = prompt(SOURCE_REGISTRY_MISSING)
-            except NoTTYException:
-                raise CLIError(NO_TTY_ERROR)
-        if not is_valid_resource_id(source_registry):
-            registry = get_registry_from_name_or_login_server(cmd.cli_ctx, source_registry, source_registry)
-            if registry:
-                source_registry = registry.id
-        source = ImportSource(resource_id=source_registry, source_image=source_image)
+    if ACR:
+        source_registry_login_server = ACR.group()
+        source_image = source[len(source_registry_login_server) + 1:]
     else:
-        source_registry_login_server = source_image[:slash]
-        source_image = source_image[slash + 1:]
-        if not source_image or not source_registry_login_server:
-            raise CLIError(INVALID_SOURCE_IMAGE)
-        registry = get_registry_from_name_or_login_server(cmd.cli_ctx, source_registry_login_server)
-        if registry:
-            if source_registry and \
-               source_registry.lower() != registry.id.lower() and \
-               source_registry.lower() != registry.name.lower() and \
-               source_registry.lower() != registry.login_server.lower():
+        slash = source.find('/')
+        source_registry_login_server = source[:slash]
+        source_image = source[slash + 1:]
+
+    registry = get_registry_from_name_or_login_server(cmd.cli_ctx, source_registry_login_server, source_registry)
```

Comment:
What if `source_registry` is a resource id?

Reply:
get_registry_from_name_or_login_server will return none

Reply:
Why, `source_registry_login_server` might find a match

Reply:
A longer answer. source_registry_login_server() searches with two filters, i.e. name and login server and only returns a valid object when exactly one registry is found. If source_registry is a resource id, it is not possible for any azure container registry to have either a name or a login server of the length of a resource ID because registry name must be between 5 to 50 characters. If source_registry_login_server is valid and it refers to a registry within the current subscription, it can return valid.